src/lib/ngsi/ContextAttribute.cpp

```diff
@@ -317,9 +415,29 @@ std::string ContextAttribute::render
   {
     if (omitValue == false)
     {
-      out += valueTag(indent + "  ", ((ciP->outFormat == XML)? "contextValue" : "value"),
-                      (request != RtUpdateContextResponse)? value : "",
-                      ciP->outFormat, commaAfterContextValue);
+      if ((valueType == ValueTypeString) || (ciP->apiVersion != "v2"))
+      {
+        out += valueTag(indent + "  ", ((ciP->outFormat == XML)? "contextValue" : "value"),
+                        (request != RtUpdateContextResponse)? stringValue : "",
+                        ciP->outFormat, commaAfterContextValue);
+      }
+#if 0
```

Comment:
The render function is only used in v1, so yes, this can be removed.
[ For v2 we use the toJson method instead ]